alf-api/src/main/java/io/axway/alf/Arguments.java

```diff
@@ -14,4 +15,13 @@
      * @return Current argument collector
      */
     Arguments add(String key, @Nullable Object value);
+
+    /**
+     * Add nested arguments
+     *
+     * @param key argument key
+     * @param arguments Nested arguments
+     * @return Current argument collector
+     */
+    Arguments add(String key, @Nullable Consumer<Arguments> arguments);
```

Comment:
should we really support null arguments consumer ? consumer are not a "value", and should never be null

Reply:
For some reason when you explicitly pass `null` this method is used instead of the `Object` overload.
Moreover renaming this method to something like `addNested` you have plenty of code to update because internally we pass `Consumer<Arguments>` a lot.
So it's `@Nullable` but it's OK. That being said I just spot a bug in the implementation of `ArgumentSerializationHelper`